src/ray/raylet/node_manager.cc

```diff
@@ -2957,16 +2968,47 @@ void NodeManager::ProcessSubscribePlasmaReady(
 
   auto message = flatbuffers::GetRoot<protocol::SubscribePlasmaReady>(message_data);
   ObjectID id = from_flatbuf<ObjectID>(*message->object_id());
-  {
-    absl::MutexLock guard(&plasma_object_notification_lock_);
-    if (!async_plasma_objects_notification_.contains(id)) {
-      async_plasma_objects_notification_.emplace(
-          id, absl::flat_hash_set<std::shared_ptr<WorkerInterface>>());
-    }
 
-    // Only insert a worker once
-    if (!async_plasma_objects_notification_[id].contains(associated_worker)) {
-      async_plasma_objects_notification_[id].insert(associated_worker);
+  if (task_dependency_manager_.CheckObjectLocal(id)) {
```

Comment:
The semantics of this method are now a bit more complicated (and important as evidenced by the issue we're solving). Could you makes sure to update the header comment for `ProcessSubscribePlasmaReady` to be more clear?